var/spack/repos/builtin/packages/py-jupyter-notebook/package.py

```diff
@@ -24,6 +25,7 @@ class PyJupyterNotebook(PythonPackage):
     variant('terminal', default=False, description="Enable terminal functionality")
 
     depends_on('python@2.7:2.8,3.3:')
```

Comment:
```suggestion
    depends_on('python@2.7:2.8,3.3:', type=('build', 'run'))
    depends_on('python@3.5:', when='@6:', type=('build', 'run'))
```
It's probably worth adding the latest version of Notebook 5.X so we have a more up-to-date version for Python 2 users.